Sprint-1/implement/sum.test.js

```diff
@@ -6,31 +6,48 @@ E.g. sum([10, 20, 30]), target output: 60
 E.g. sum(['hey', 10, 'hi', 60, 10]), target output: 80 (ignore any non-numerical elements)
 */
 
-const sum = require("./sum.js");
+const sum = require("./sum");
 
 // Acceptance Criteria:
 
 // Given an empty array
 // When passed to the sum function
 // Then it should return 0
-test.todo("given an empty array, returns 0")
+test("Given an empty array, it should return 0", () => {
+  expect(sum([])).toBe(0);
+});
 
 // Given an array with just one number
 // When passed to the sum function
 // Then it should return that number
+test("Given an array with just one number, it should return that number", () => {
+  expect(sum([10])).toBe(10);
+});
 
 // Given an array containing negative numbers
 // When passed to the sum function
 // Then it should still return the correct total sum
+test("Given an array containing negative numbers, it should return the correct total sum", () => {
+  expect(sum([10, -5, -2, 5])).toBe(8);
+});
 
 // Given an array with decimal/float numbers
 // When passed to the sum function
 // Then it should return the correct total sum
+test("Given an array with decimal/float numbers, it should return the correct total sum", () => {
+  expect(sum([10.5, 20.3, 30.2])).toBe(61);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?

Reply:
I believe it's better to update the function itself rather than adjusting the test. To handle floating-point precision issues, I’ve modified the return statement to round the sum to 10 decimal places using `Math.round(sum * 1e10) / 1e10`

I can update the test to use : `expect(Math.abs(sum([1.1, 1.1, 1.1]) - 3.3)).toBeLessThan(0.0001);`
for more accurate comparison.